tests/cpp_unit_tests/test_asym_line.cpp

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include <power_grid_model/component/asym_line.hpp>
+
+#include <doctest/doctest.h>
+
+namespace power_grid_model {
+
+using namespace std::complex_literals;
+
+TEST_CASE("Test asym line") {
+    double system_frequency = 50.0;
+    AsymLineInput const input{.id = 1,
+                          .from_node = 2,
+                          .to_node = 3,
+                          .from_status = 1,
+                          .to_status = 1,
+                          .r_aa = 0.4369,
+                          .r_ba = 0.0496,
+                          .r_bb = 0.4369,
+                          .r_ca = 0.0485,
+                          .r_cb = 0.0496,
+                          .r_cc = 0.4369,
+                          .r_na = 0.0496,
+                          .r_nb = 0.0485,
+                          .r_nc = 0.0496,
+                          .r_nn = 0.4369,
+                          .x_aa = 0.8538,
+                          .x_ba = 0.7886, 
+                          .x_bb = 0.8538, 
+                          .x_ca = 0.7663, 
+                          .x_cb = 0.7886,
+                          .x_cc = 0.8538,
+                          .x_na = 0.7886, 
+                          .x_nb = 0.7663,
+                          .x_nc = 0.7886,
+                          .x_nn = 0.8538,
+                          .c0 = 0.18,
+                          .c1 = 0.308,
+                          .i_n = 216.0};
+    AsymLine asym_line{input, system_frequency, 10.0e3, 10.0e3};
+    double const base_i = base_power_1p / (10.0e3 / sqrt3);
+    double const base_y = base_i * base_i / base_power_1p;
+    Branch& branch = asym_line;
+    ComplexTensor<asymmetric_t> const y_series = ComplexTensor<asymmetric_t>(1.87842984-0.42269873i, 1.87842984-0.42269873i, 1.87842984-0.42269873i, -0.62560863-0.00463073i, -0.57187623+0.12931409i, -0.62560863-0.00463073i);
```

Comment:
where do you get those numbers? maybe you want to calculate them based on `input` instead of hardcode the number. The future developer can understand the logic easier.